scripts/check_go_version.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+MOD_VERSION=$(go mod edit -json | jq -r .Go)
+echo "go.mod version: $MOD_VERSION"
+
+for wf in .github/workflows/*.yml; do
+    echo "Checking $wf ..."
+
+    WF_VERSIONS=$(yq -r '.jobs[].steps[] | select(.with["go-version"]) | .with["go-version"]' -o=tsv "$wf" | grep -v '^---$' || true)
+    if [ -z "$WF_VERSIONS" ]; then
+        echo "ℹ️ No go-version found in $wf (skipped)"
+        continue
+    fi
+
+    UNIQUE_WF_VERSIONS=$(echo "$WF_VERSIONS" | sort -u)
+    if [ "$(echo "$UNIQUE_WF_VERSIONS" | wc -l)" -ne 1 ]; then
+        echo "❌ Multiple Go versions found in $wf:"
+        echo "$UNIQUE_WF_VERSIONS"
+        exit 1
```

Comment:
We could gather up all errors from all workflows and print or auto-correct everything at the end? It'll be annoying to run this script, get one error, fix, get another, etc.

Reply:
I'm hesitant about auto-correcting this given that it's meant to be run as a CI step. Maybe an optional argument `--autofix`?

Reply:
Partially addressed in 759a03e, auto-fixing may be a separate change